### PR TITLE
Default to `HEAD` on `send-comment --commit-sha`

### LINKED ISSUE #991: `send-comment --pr --update` not working without specifying `--commit-sha=HEAD`
This workflow threw the following error:

```
- name: Train model
        env:
          REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
        run: |
          set -x
          cml ci
          pip install -r requirements.txt
          python get_data.py
          python train.py

          # Create pull request
          cml pr model/random_forest.joblib

          # Create CML report
          cat model/metrics.txt > report.md
          cml publish model/confusion_matrix.png --md >> report.md
          cml send-comment --pr --update report.md
```
```
{"level":"error","message":"PR for commit sha \"54e32ca98b5faa1f7982a682e0f6de07347a7531\" not found","stack":"Error: PR for commit sha \"54e32ca98b5faa1f7982a682e0f6de07347a7531\" not found\n    at CML.commentCreate (/snapshot/cml/src/cml.js:185:15)\n    at async Object.exports.handler (/snapshot/cml/bin/cml/send-comment.js:13:15)"}
Error: Process completed with exit code 1.
```

After some back and forth on Slack, we came to the following:

:x: Do not work:
- `cml send-comment --pr --update report.md`
- `cml send-comment --commit-sha --pr --update report.md`

:white_check_mark: Do work:
- `cml send-comment --update report.md`
- `cml send-comment --commit-sha=HEAD --pr --update report.md`

I am currently unsure of the precise intended workings; is `send-comment --pr --update` a valid use case or not? Different views in Slack, so creating a ticket here for further discussion.

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
> [_`--commit-sha=<rev>`: Git revision linked to this comment [default: HEAD]._](https://cml.dev/doc/ref/send-comment#options)

Closes #991. Documentation is a lie! It defaulted to `triggerSha` which doesn't get updated when `cml pr` runs.

https://github.com/iterative/cml/blob/352fba44faf2eb69d81d4cb5f60220f60f5f8e49/src/cml.js#L143

### CODE REVIEW

**Review by DavidGOrtega — CHANGES_REQUESTED** [review]

File: bin/cml/send-comment.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ exports.builder = (yargs) =>
       commitSha: {
         type: 'string',
         alias: 'head-sha',
-        description: 'Commit SHA linked to this comment. Defaults to HEAD.'
+        default: 'HEAD',
```
[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply bin/cml/send-comment.js:27]: Not an issue? See https://github.com/iterative/cml/pull/790#issuecomment-954655519

[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply bin/cml/send-comment.js:27]: Yes, [to my mind] you were absolutely right. Commenting on `triggerSha` after `cml pr` is pointless and [astonishing](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

[the remark this answers is not in the document]
Comment (casperdcl) [reply bin/cml/send-comment.js:27]: What about non-PR use cases?

```bash
local$ git commit -m experiment && git push
ci-cd$ dvc repro && git add -am "experiment results" && git push
ci-cd$ cml send-comment # probably ok to change from triggerSha to HEAD
```

**Review by casperdcl — APPROVED** [review]

File: bin/cml/send-comment.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply bin/cml/send-comment.js:27]: Again, the “expected” behavior, if any, would be commenting on the current commit, not on the triggering one. 🤔 

Even if it's a “breaking” change for some workflows, nobody is going to notice the difference.

**Pull request merged**